### PR TITLE
Rework pillar to work with terragrunt

### PR DESCRIPTION
**Pull request by confusdcodr**
terragrunt issue 829

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -82,12 +82,50 @@ data "aws_subnet" "lb" {
   id = var.lb_subnet_ids[count.index]
 }
 
+# Sorting to alernative means of creating the directory due to
+# https://github.com/gruntwork-io/terragrunt/issues/829
 # Manage vault pillar
-resource "template_dir" "pillar" {
-  source_dir      = var.vault_pillar_path
-  destination_dir = "${path.cwd}/.files/pillar"
+#resource "template_dir" "pillar" {
```
Comment (userhas404d) [inline main.tf]: any particular reason for this comment block?

Comment (confusdcodr) [reply main.tf]: Was leaving it in there so that we could revert to this methodology if the issue were to be resolved.

Comment (lorengordon) [reply main.tf]: Git will show us the old method. It is sufficient to keep a comment mentioning why template_dir was not used and link to the issue. We do not keep crufty code in git.

File: main.tf [hunk 2]
```diff
@@ -82,12 +82,50 @@ data "aws_subnet" "lb" {
   id = var.lb_subnet_ids[count.index]
 }
 
+# Sorting to alernative means of creating the directory due to
```
Comment (lorengordon) [inline main.tf]:
```suggestion
# Resorting to alternative means of creating the directory due to
```

File: main.tf [hunk 3]
```diff
@@ -82,12 +82,13 @@ data "aws_subnet" "lb" {
   id = var.lb_subnet_ids[count.index]
 }
 
-# Manage vault pillar
-resource "template_dir" "pillar" {
-  source_dir      = var.vault_pillar_path
-  destination_dir = "${path.cwd}/.files/pillar"
+# Resorting to alernative means of creating the directory due to
```
Comment (lorengordon) [inline main.tf]:
```suggestion
# Resorting to alternative means of creating the directory due to
```

**Pull request merged**